src/frontend/components/StudentLiveControlButtons/StudentJoinDiscussionButton/index.spec.tsx

```diff
@@ -0,0 +1,257 @@
+import {
+  act,
+  fireEvent,
+  render,
+  screen,
+  waitFor,
+} from '@testing-library/react';
+import React from 'react';
+import { Toaster } from 'react-hot-toast';
+
+import { PUBLIC_JITSI_ROUTE } from 'components/PublicVideoLiveJitsi/route';
+import { useParticipantWorkflow } from 'data/stores/useParticipantWorkflow';
+import { wrapInIntlProvider } from 'utils/tests/intl';
+import { wrapInRouter } from 'utils/tests/router';
+import { converse } from 'utils/window';
+
+import { StudentJoinDiscussionButton } from '.';
+
+//  matchMedia is not defined in tests
+//  https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
+Object.defineProperty(window, 'matchMedia', {
+  writable: true,
+  value: jest.fn().mockImplementation((query) => ({
+    matches: false,
+    media: query,
+    onchange: null,
+    addListener: jest.fn(), // Deprecated
+    removeListener: jest.fn(), // Deprecated
+    addEventListener: jest.fn(),
+    removeEventListener: jest.fn(),
+    dispatchEvent: jest.fn(),
+  })),
+});
```

Comment:
Please see components/PlaylistPortability/index.spec.tsx where `jest-matchmedia-mock` is used.